sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/TokenResult.java

```diff
@@ -107,7 +107,9 @@ public TokenResult withExpirationTime(Date expiresOn) {
         try {
             this.expiresOn = JSONDateUtils.toString(expiresOn);
         } catch (JSONException ex) {
-            AppCenterLog.error(Constants.LOG_TAG, "Unable to convert '" + expiresOn.toString() + "' Date to ISO 8601 string");
+            AppCenterLog.error(Constants.LOG_TAG, "Unable to convert '" +
+                    ((expiresOn == null) ? "(null)": expiresOn.toString()) +
```

Comment:
toString() no needed, please also remove parentheses around expiresOn == null, you should have a grey highlight on Android Studio for that one.
Overall, it would be better to use a string format or use local variables to improve formatting here.